packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -203,3 +257,49 @@ class _BottomAppBarClipper extends CustomClipper<Path> {
         || oldClipper.notchMargin != notchMargin;
   }
 }
+
+class _DefaultsM2 extends BottomAppBarTheme {
+  const _DefaultsM2(this.context)
```

Comment:
Here again, the new convention is:

```suggestion
class _BottomAppBarDefaultsM2 extends BottomAppBarTheme {
  const _BottomAppBarDefaultsM2(this.context)
```